app/scripts/components/criteria/variables.service.js

```diff
@@ -132,32 +129,32 @@ angular.module("chuvApp.components.criteria").factory("Variable", [
         return breadcrumb;
       });
 
-    resource.getVariableData = variableCode =>
-      resource.hierarchy().then(
-        data =>
```

Comment:
here sorry but js Promises have to go because they do not trigger a digest like $q does

Reply:
We need 
```
const groups = current.groups || [];
const variables = current.variables || [];
```